### PR TITLE
Feat/office prepaid bc

### PR DESCRIPTION
**Pull request by euhmeuh**
Allow prepaid office users to order new prepaid licenses.

### CODE REVIEW

File: src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+angular.module("Module.microsoft.controllers").controller("MicrosoftOfficeLicenseUserOrderCtrl", class MicrosoftOfficeLicenseUserOrderCtrl {
+
+    constructor (Alerter, MicrosoftOfficeLicenseService, $rootScope, $scope, OvhHttp) {
+        this.alerter = Alerter;
+        this.licenseService = MicrosoftOfficeLicenseService;
+        this.$rootScope = $rootScope;
+        this.$scope = $scope;
+        this.OvhHttp = OvhHttp;
+
+        this.licenseId = this.$scope.currentActionData.license;
```
Comment (FredericEspiau) [inline src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js]: => `$onInit()`

File: src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    $onInit () {
+        this.loaders = {
+            licenseEnum: false,
+            userOrder: false
+        };
+
+        this.license = null;
+        this.number = 1;
```
Comment (FredericEspiau) [inline src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js]: `number` of what ?

File: src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    orderUser () {
+        this.loaders.userOrder = true;
+        this.licenseService.gotToOrderPrepaidLicenses(this.licenseId, this.license, this.number);
+    }
+
+    getLicenses (serviceName) {
+        this.loaders.licenseEnum = true;
+
+        return this.OvhHttp.get(`/order/cartServiceOption/office365Prepaid/${serviceName}`, {
```
Comment (FredericEspiau) [inline src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js]: should be in a service

File: src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        this.number = 1;
+
+        this.$scope.orderUser = () => this.orderUser();
+
+        this.getLicenses(this.licenseId);
+    }
+
+    orderUser () {
+        this.loaders.userOrder = true;
+        this.licenseService.gotToOrderPrepaidLicenses(this.licenseId, this.license, this.number);
```
Comment (FredericEspiau) [inline src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js]: The service changes the window or opens a new tab ? I don't know about that, what do you think @jleveugle ?

File: src/microsoft/office/license/user/order/microsoft-office-license-user-order.html [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        <div data-wizard-step
+             data-wizard-step-valid="orderUserCtrl.number && orderUserCtrl.license">
+
+            <form class="form-horizontal mt-4" name="microsoftOfficeLicenseOrderUserForm">
+                <div class="form-group"
+                     data-ng-class="{'has-error': microsoftOfficeLicenseOrderUserForm.number.$dirty && microsoftOfficeLicenseOrderUserForm.number.$invalid}">
+                    <label class="control-label col-md-4 required" for="number"
+                           data-i18n-static="microsoft_office_license_order_user_number_label"></label>
+                    <div class="col-md-6">
+                        <input type="number" class="form-control" id="number" name="number" min="0" required
```
Comment (FredericEspiau) [inline src/microsoft/office/license/user/order/microsoft-office-license-user-order.html]: `min` should be backed up by`data-ng-min` as `min` is not supported by older browsers

File: src/microsoft/office/license/user/order/microsoft-office-license-user-order.html [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+                    </div>
+                </div>
+                <div class="form-group">
+                    <label class="control-label col-md-4 required" for="license"
+                           data-i18n-static="microsoft_office_license_order_user_license_label"></label>
+                    <div class="col-md-6">
+                        <div class="oui-select">
+                            <select class="oui-select__input" id="license" name="license" required
+                                    data-ng-model="orderUserCtrl.license"
+                                    data-ng-options="licenseType as tr('microsoft_office_license_'+licenseType) for licenseType in orderUserCtrl.licenseEnum track by licenseType">
```
Comment (FredericEspiau) [inline src/microsoft/office/license/user/order/microsoft-office-license-user-order.html]: `tr('microsoft_office_license_' + licenseType)` is a bit more readable ;)

File: src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    orderUser () {
+        this.loaders.userOrder = true;
+        this.licenseService.gotToOrderPrepaidLicenses(this.licenseId, this.license, this.numberOfLicenses);
+    }
+
+    getLicenses (serviceName) {
+        this.loaders.licenseEnum = true;
+
+        return this.licenseService.getAvailableOptions(serviceName)
+            .then((licenses) => {
```
Comment (jleveugle) [inline src/microsoft/office/license/user/order/microsoft-office-license-user-order.controller.js]: catch() ? :)

File: bower.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ovh-module-office",
-  "version": "3.0.1",
```
Comment (antleblanc) [inline bower.json]: Wrong merge?

File: package.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ovh-module-office",
-  "version": "3.0.1",
```
Comment (antleblanc) [inline package.json]: Wrong merge?

File: src/microsoft/office/license/user/order/microsoft-office-license-user-order.html [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+            </form>
+        </div>
+
+        <div data-wizard-step>
+            <p data-i18n-static="microsoft_office_license_order_user_confirm"></p>
+            <dl class="dl-horizontal dl-lg">
+                <dt data-i18n-static="microsoft_office_license_order_user_number_label"></dt>
+                <dd data-ng-bind="orderUserCtrl.numberOfLicenses"></dd>
+            </dl>
+            <dl class="dl-horizontal dl-lg">
```
Comment (antleblanc) [inline src/microsoft/office/license/user/order/microsoft-office-license-user-order.html]: You can use a single `<dl>`.

File: src/resources/i18n/microsoft/Messages_fr_FR.xml [hunk 11]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <translations>
 
-   <translation id="microsoft_office_license_officeProPlusofficeBusiness" qtlid="253218">Office Business</translation>
-   <translation id="microsoft_office_license_officeProPlusofficeProPlus" qtlid="253231">Office Pro Plus</translation>
+   <translation id="microsoft_office_license_office-business-pp" qtlid="253218">Office Business</translation>
+   <translation id="microsoft_office_license_office-proplus-pp" qtlid="253231">Office Pro Plus</translation>
+   <translation id="microsoft_office_license_officeBusiness" qtlid="253218">Office Business</translation>
```
Comment (antleblanc) [inline src/resources/i18n/microsoft/Messages_fr_FR.xml]: Duplicates.

Comment (euhmeuh) [reply src/resources/i18n/microsoft/Messages_fr_FR.xml]: I know... but if you find a better way...
One is fetched from licenseEnum in the json schema, the other is fetched from the API response.

File: package.json (hunk 9 above)
Comment (euhmeuh) [reply package.json]: Rebase done.

File: bower.json (hunk 8 above)
Comment (euhmeuh) [reply bower.json]: Rebase done.

**Pull request merged**